pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -309,19 +309,8 @@ impl WalreceiverState {
 
         let id = self.id;
         let connect_timeout = self.wal_connect_timeout;
-        let connection_attempt = self
-            .wal_connection_attempts
-            .get(&new_sk_id)
-            .copied()
-            .unwrap_or(0);
         let connection_handle = TaskHandle::spawn(move |events_sender, cancellation| {
             async move {
-                exponential_backoff(
-                    connection_attempt,
-                    DEFAULT_BASE_BACKOFF_SECONDS,
```

Comment:
Without having backoff we can easily overflow safekeepers with endless connections in a loop. Or do we have sleep in some other place to prevent that?

I would keep backoff here for a while and just changed `DEFAULT_BASE_BACKOFF_SECONDS` to `0.1s` and `DEFAULT_MAX_BACKOFF_SECONDS` to `3s`. WDYT?

Reply:
Yes, this way it indeed seems better after I've tried it a bit, thank you for proposing.